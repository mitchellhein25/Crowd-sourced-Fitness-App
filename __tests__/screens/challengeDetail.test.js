import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import ChallengeDetail from '../../src/screens/challengeDetail';

describe('<ChallengeDetail /> load', () => {
    beforeAll(() => {
        item = {
            id: 'challengeTestId',
            description: '',
            date: '',
            type: 'RUNNING',
            goals: [],
            tags: [''],
            badges: ['BBStrength']
        }

        userId = 'testId'

        route = {
            params: {
                challenge: item,
                userId: userId
            }
        }
    });
    
    it('has 7 children', () => {
        const tree = renderer.create(<ChallengeDetail route={route} />).toJSON();
        expect(tree.children.length).toBe(7);
    });

    it('renders correctly', async () => {
        const tree = renderer.create(<ChallengeDetail route={route} />).toJSON();
        await act(async () => { await expect(tree).toMatchSnapshot(); })
    });
});

