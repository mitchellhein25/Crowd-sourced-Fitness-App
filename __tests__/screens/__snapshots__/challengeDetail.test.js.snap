// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChallengeDetail /> load renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 20,
      "paddingTop": "15%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "black",
        "borderRadius": 30,
        "margin": 5,
        "padding": 8,
        "width": 150,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 20,
            "paddingLeft": 10,
          }
        }
      >
        Go Back
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "500",
      }
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 25,
        "marginTop": 20,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "500",
        }
      }
    >
      Type:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      

      Running
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 25,
        "marginTop": 20,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "500",
        }
      }
    >
      Goals:
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 25,
        "marginTop": 20,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "500",
        }
      }
    >
      Badges:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      

      Barbell Strength
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 25,
        "marginTop": 20,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "500",
        }
      }
    >
      Tags:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      

      
    </Text>
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "black",
        "borderRadius": 30,
        "borderWidth": 1,
        "marginTop": 20,
        "padding": 8,
        "width": 150,
      }
    }
  >
    <View
      accessibilityLabel="Join this Challenge button"
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          Join this Challenge
        </Text>
      </View>
    </View>
  </View>
</View>
`;
